### PR TITLE
*: use syncutil lock

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #4399

### What is changed and how does it work?

Replace sync.RWMutex/Mutex with syncutil which will be changed to deadlock one in tests.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/balancer/round_robin.go [hunk 1]
```diff
@@ -15,13 +15,14 @@
...
-	"sync"
 	"sync/atomic"
+
+	"github.com/tikv/pd/pkg/utils/syncutil"
 )
 
 // RoundRobin is a balancer that selects nodes in a round-robin fashion.
 type RoundRobin[T uint32 | string] struct {
-	sync.RWMutex
+	syncutil.RWMutex
```
Comment (lhy1024) [inline pkg/balancer/round_robin.go:25]: What difference are there?

Comment (rleungx) [reply pkg/balancer/round_robin.go:25]: If we run `make test`, it will use the deadlock RWMutex.

**Review by lhy1024 — APPROVED** [review]

**Review by HuSharp — COMMENTED** [review]
I noticed a few `sync.RWMutex`s have not been changed, does this make sense?
For example, in `keyspaceGroupSvcDiscovery`

**Comment by rleungx** [comment]
> I noticed a few `sync.RWMutex`s have not been changed, does this make sense? For example, in `keyspaceGroupSvcDiscovery`

They are in client module

**Review by HuSharp — APPROVED** [review]
LGTM!

**Comment by HuSharp** [comment]
> > I noticed a few `sync.RWMutex`s have not been changed, does this make sense? For example, in `keyspaceGroupSvcDiscovery`
> 
> They are in client module

get

**Pull request merged**